mgmtsystem/azure.py

```diff
@@ -182,8 +189,27 @@ def list_flavor(self):
     def list_network(self):
         raise NotImplementedError('list_network not implemented.')
 
-    def vm_creation_time(self, vm_name):
-        raise NotImplementedError('vm_creation_time not implemented.')
+    def vm_creation_time(self, vm_name, resource_group=None):
+        self.logger.info("Attempting to Retrieve Azure VM Modification Time {}".format(vm_name))
+        vm_vhd = self.get_vm_vhd(vm_name, resource_group or self.resource_group)
+        vhd_name = os.path.split(urlparse.urlparse(vm_vhd).path)[1]
+        data = self.run_script(
+            """
+            Invoke-Command -scriptblock {{
+            $storageContext = New-AzureStorageContext -StorageAccountName \"{storage_account}\" `
+                            -StorageAccountKey \"{storage_key}\"
+            Get-AzureStorageBlob -Name \"{storage_blob}\" `
+                            -Context $storageContext -Blob \"{vhd_blob}\" | convertto-xml -as String
+            }}
+            """.format(storage_account=self.storage_account,
+                       storage_blob=self.storage_blob,
+                       storage_key=self.storage_key,
+                       vhd_blob=vhd_name), True)
+        vhd_last_modified = etree.parse(StringIO(self.clean_azure_xml(data))).getroot().xpath(
+            "./Object/Property[@Name='LastModified']/text()")
+        create_time = datetime.strptime(str(vhd_last_modified), '[\'%m/%d/%Y %H:%M:%S %p +00:00\']')
+        self.logger.info("VM last edit time based on vhd =  {}".format(str(create_time)))
```

Comment:
Yeah, there is no such thing as vm creation time in azure.  The best Amogh and I could come up with is VHD last modification.